packages/w/wayland-protocols/rules/wayland_protocols.lua

```diff
@@ -14,67 +14,122 @@
         end)
     end
 ]]
+-- Options:
+--[[
+    add_rules("@wayland-protocols/wayland.protocols", {
+        outputdir = "...",  -- Path for generated files, default is `path.join(target:autogendir(), "rules", "wayland.protocols")`
+        client = "...",     -- Path format for client protocol header files, default is `"%s.h"`, set to `false` to disable its generation
+        server = "...",     -- Path format for server protocol header files, default is `nil`
+        code = "...",       -- Path format for code files, default is `"%s.c"`
+        public = false,     -- Visibility for headers and symbols in generated code, can be `false` or `true`, default is `false`
+    })
+]]
+
 rule("wayland.protocols")
     set_extensions(".xml")
 
-    on_load(function(target)
+    on_config(function(target)
+        import("core.base.option")
+
+        local rule_name = "@wayland-protocols/wayland.protocols"
+
         if target:rule("c++.build") then
             local rule = target:rule("c++.build"):clone()
-            rule:add("deps", "wayland.protocols", {order = true})
+            rule:add("deps", rule_name, {order = true})
             target:rule_add(rule)
         end
-    end)
 
-    on_config(function(target)
-        import("core.base.option")
+        local public = target:extraconf("rules", rule_name, "public")
 
-        local outputdir = target:extraconf("rules", "wayland.protocols", "outputdir") or path.join(target:autogendir(), "rules", "wayland.protocols")
+        local outputdir = target:extraconf("rules", rule_name, "outputdir") or path.join(target:autogendir(), "rules", "wayland.protocols")
         if not os.isdir(outputdir) then
             os.mkdir(outputdir)
         end
-        target:add("includedirs", outputdir)
+        target:add("includedirs", outputdir, {public = public})
 
         local dryrun = option.get("dry-run")
-        local sourcebatches = target:sourcebatches()
-        if not dryrun and sourcebatches["wayland.protocols"] and sourcebatches["wayland.protocols"].sourcefiles then
-            for _, protocol in ipairs(sourcebatches["wayland.protocols"].sourcefiles) do
-                local clientfile = path.join(outputdir, path.basename(protocol) .. ".h")
-                local privatefile = path.join(outputdir, path.basename(protocol) .. ".c")
-
-                -- for c++ module dependency discovery
-                if not os.exists(clientfile) then
-                    os.touch(clientfile)
+        local sourcebatches = target:sourcebatches()[rule_name]
+        if not dryrun and sourcebatches and sourcebatches.sourcefiles then
+            local client = target:extraconf("rules", rule_name, "client")
+            if client == nil then
+                client = "%s.h"
+            end
```

Comment:
```
local client = target:extraconf("rules", rule_name, "client") or "%s.h"
```

Reply:
I would like to use `false` in this option in order to disable the generation of client headers here (e.g. if one only wants to generate server headers), so we cannot simply write `or "%s.h"` here.